torch/csrc/inductor/aoti_torch/opaque_tensor.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
```

Comment:
Rename this file into something more meaningful, like `mkldnn_tensor.*`. 

Also is this ever useful to other components beyond AOTI? If yes, please move it to some other common directory.

Reply:
I've renamed the files to `mkldnn_tensor.*` and moved the actual implementation under `aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp`. The `mkldnn_tensor.*` files are kept to handle the cases both MKL-DNN is enabled or not. Let me know if more refine is needed.